.github/workflows/release.yml

```diff
@@ -1,23 +1,69 @@
-name: Release policy server
+name: policy-server release
 on:
-  workflow_run:
-    workflows: ["Build policy server"]
-    types:
-      - completed
-    branches:
-      - "v*"
+  push:
+    tags:
+    - 'v*'
+
+# Declare default permissions as read only.
+permissions: read-all
 
 jobs:
+  ci:
+    uses: ./.github/workflows/ci.yml
+    permissions: read-all
+
+  container-build:
+    uses: ./.github/workflows/container-build.yml
+    permissions:
+      id-token: write
+      packages: write
   release:
-    name: Create release
     permissions:
+      id-token: write
       contents: write
+    name: Create release
     runs-on: ubuntu-latest
-    if: ${{ github.event.workflow_run.conclusion == 'success' }}
+    needs:
+      - ci
+      - container-build
     steps:
+      - name: Install Golang
+        uses: actions/setup-go@v4
+        with:
+          go-version: '1.20'
+
+      - name: Install the bom command
+        shell: bash
+        run: go install sigs.k8s.io/bom/cmd/bom@v0.2.2
+
+      - name: Install cosign
+        uses: sigstore/cosign-installer@v3
+
+      - name: Checkout code
+        uses: actions/checkout@v3
+
+      - name: Retrieve tag name
+        if: ${{ startsWith(github.ref, 'refs/tags/') }}
+        run: |
+          echo TAG_NAME=$(echo ${{ github.ref_name }}) >> $GITHUB_ENV
+
+      - name: Create SBOM file
+        shell: bash
+        run: |
+          bom generate -n https://kubewarden.io/kubewarden.spdx \
+            --image "ghcr.io/${{github.repository_owner}}/policy-server@${{ needs.container-build.outputs.digest }}" \
+            . > policy-server-sbom.spdx
+
+      - name: Sign BOM file
+        run: |
+          cosign sign-blob --yes --output-certificate policy-server-sbom.spdx.cert \
+            --output-signature policy-server-sbom.spdx.sig \
+            policy-server-sbom.spdx
+        env:
+          COSIGN_EXPERIMENTAL: 1
```

Comment:
```suggestion
```

We do not need this. Keyless signing is not experimental anymore. 

https://blog.sigstore.dev/cosign-2-0-released/